### PR TITLE
Always export related images when a CSV has spec.relatedImages

### PR DESCRIPTION
**Pull request by tkdchen**
* CLOUDBLD-2244

Signed-off-by: Chenxiong Qi <cqi@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- n/a - JSON/YAML configuration changes are updated in the relevant schema
- n/a - Changes to metadata also update the documentation for the metadata
- n/a - Pull request has a link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/plugins/pre_pin_operator_digest.py [hunk 1]
```diff
@@ -196,7 +214,9 @@ def _get_pullspecs(self, operator_csv):
                    "automatically.".format(operator_csv.path))
             raise RuntimeError(msg)
         else:
-            self.log.warning("%s has a relatedImages section, skipping", operator_csv.path)
+            related_images = operator_csv._related_image_pullspecs()
+            self.log.info("%s has a relatedImages section, skipping", operator_csv.path)
```
Comment (MartinBasti) [inline atomic_reactor/plugins/pre_pin_operator_digest.py:218]: Maybe that `skipping` part should be removed, because we are processing pullspecs

Comment (tkdchen) [reply atomic_reactor/plugins/pre_pin_operator_digest.py]: Removed.

File: atomic_reactor/plugins/pre_pin_operator_digest.py [hunk 2]
```diff
@@ -196,7 +214,8 @@ def _get_pullspecs(self, operator_csv):
                    "automatically.".format(operator_csv.path))
             raise RuntimeError(msg)
         else:
-            self.log.warning("%s has a relatedImages section, skipping", operator_csv.path)
+            related_images = operator_csv._related_image_pullspecs()
```
Comment (chmeliik) [inline atomic_reactor/plugins/pre_pin_operator_digest.py:217]: This method is internal to the operator module, as is the return value (list of `NamedPullspec` objects). Should the there be a public `get_related_image_pullspecs()` method which returns a set of `ImageName` objects?

Comment (tkdchen) [reply atomic_reactor/plugins/pre_pin_operator_digest.py]: Thanks @chmeliik. Updated.

Comment (chmeliik) [reply atomic_reactor/plugins/pre_pin_operator_digest.py]: IMO the public method should return a collection of `ImageName`, not `NamedPullspec`. `NamedPullspec` was intended to be a private part of the operator module. I think the `_related_image_pullspecs` method should be kept and `get_related_image_pullspecs` should be a new method.

**Comment by rcerven** [comment]
code looks ok

just that comment about _related_image_pullspecs vs related_image_pullspecs methods has to be resolved

File: atomic_reactor/plugins/pre_pin_operator_digest.py (hunk 2 above)
Comment (tkdchen) [reply atomic_reactor/plugins/pre_pin_operator_digest.py]: @chmeliik PTAL

Comment (chmeliik) [reply atomic_reactor/plugins/pre_pin_operator_digest.py]: Looks good :+1:

**Review by MartinBasti — APPROVED** [review]

File: atomic_reactor/utils/operator.py [hunk 3]
```diff
@@ -431,12 +431,27 @@ def _named_pullspecs(self):
...
+        :return: a list of pullspecs. It could be an empty if no spec.RelatedImage section.
+        :rtype: list[RelatedImage]
+        """
         related_images_path = ("spec", "relatedImages")
         return [
             RelatedImage(r)
             for r in chain_get(self.data, related_images_path, default=[])
         ]
 
+    def get_related_image_pullspecs(self):
```
Comment (chmeliik) [inline atomic_reactor/utils/operator.py:445]: Could we also have a simple unit test for this in `test_operator.py`?

Comment (tkdchen) [reply atomic_reactor/utils/operator.py:445]: Unit test is added.

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

@lcarva operator module was updated, in case you want to sync this to https://github.com/containerbuildsystem/operator-manifest

**Comment by MartinBasti** [comment]
@tkdchen please add release notes (to the next release draft and here as a comment)

**Comment by MartinBasti** [comment]
ping

**Comment by tkdchen** [comment]
@MartinBasti Updated. Sorry for late.

**Pull request merged**